consensus/spos/bls/subroundSignature.go

```diff
@@ -239,3 +239,31 @@ func (sr *subroundSignature) signaturesCollected(threshold int) (bool, int) {
 
 	return n >= threshold, n
 }
+
+func (sr *subroundSignature) waitAllSignatures() {
+	for {
```

Comment:
I think this for is useless
Following Robert's suggestion the code might look like this:
```GO
func (sr *subroundSignature) waitAllSignatures(ctx context.Context) {
   remainingTime := sr.remainingTime()
   select{
       case <- ctx.Done:
            return
       case <-time.After(remainginTime): 
   }
   
   if sr.IsSubroundFinished(sr.Current()) {
        return 
   }

   select {
      case sr.ConsensusChannel() <- true:
      default:
   }

}
```

Reply:
Done